master/buildbot/buildslave/protocols/base.py

```diff
@@ -66,3 +76,39 @@ def remoteStartBuild(self, builderName):
 
     def remoteInterruptCommand(self, commandId, why):
         raise NotImplementedError
+
+
+# RemoteCommand base implementation and base proxy
```

Comment:
Well, this file was already based on subclassing.. and you mentored this gsoc project, afaik :)

If we'd followed all google's best practices, we'd rewrite buildbot in golang. :->